From 97f126b8c2969bdd2cc1245b2e9c09a9ef5ca82b Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Fri, 22 Sep 2023 10:32:01 +0200
Subject: [PATCH 0559/1141] dt-bindings: stm32: make dma-ranges optional in
 mlahb

On stm32mp25 SoCs there is no more need of memory translation between
the Cortex-A and the Cortex-M. In this case the dma-ranges is not used.

A common solution has to be implemented that matches with the stm32mp15
and stm32mp25 platforms.

The option that consists in using an empty dma-ranges is not properly
working due to the different architectures (arm and arm64) of the
processors. this generate such kind of warning:
/ahb:dma-ranges: empty "dma-ranges" property but its #address-cells (1) differs from / (2)

The solution implemented is to remove the dma-ranges property when not
memory translation.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I389438ab4ed4150baa6f39623f48ea12c0497ac9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/329140
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 Documentation/devicetree/bindings/arm/stm32/st,mlahb.yaml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/Documentation/devicetree/bindings/arm/stm32/st,mlahb.yaml
+++ b/Documentation/devicetree/bindings/arm/stm32/st,mlahb.yaml
@@ -48,7 +48,7 @@ required:
   - compatible
   - '#address-cells'
   - '#size-cells'
-  - dma-ranges
+
 
 unevaluatedProperties: false
 
